pkg/plugin/trivy/plugin.go

```diff
@@ -956,10 +956,12 @@ func (p *plugin) getPodSpecForStandaloneFSMode(ctx starboard.PluginContext, conf
 	if err != nil {
 		return corev1.PodSpec{}, nil, err
 	}
-	// get nodeName from running pods.
-	spec.NodeName, err = p.objectResolver.GetNodeName(context.Background(), workload)
-	if err != nil {
-		return corev1.PodSpec{}, nil, err
+	if !ctx.GetStarboardConfig().IsScanJobsInSameNamespace() {
+		// get nodeName from running pods.
+		spec.NodeName, err = p.objectResolver.GetNodeName(context.Background(), workload)
+		if err != nil {
+			return corev1.PodSpec{}, nil, err
```

Comment:
Let's add more context in case of failure so we will see exactly in the logs what is going on and which workload is problematic:
```suggestion
			return corev1.PodSpec{}, nil, fmt.Errorf("failed resolving node name for workload %q: %w", workload.Namespace+"/"+workload.Name, err)
```